File: app/src/main/java/com/chainreaction/mercadopagoclient/GatewayService.java
package com.chainreaction.mercadopagoclient;

import com.chainreaction.mercadopagoclient.model.Card;
import com.chainreaction.mercadopagoclient.model.Token;

import retrofit.Callback;
import retrofit.http.Body;
import retrofit.http.POST;
import retrofit.http.Query;

public interface GatewayService {
    @POST("/card_tokens")
    void getToken(@Query("public_key") String public_key, @Body Card card, Callback<Token> callback);

    @POST("/card_tokens")
    Token getToken(@Query("public_key") String public_key, @Body Card card);
}

File: app/src/main/java/com/chainreaction/mercadopagoclient/GatewayService.kt
package com.chainreaction.mercadopagoclient

import com.chainreaction.mercadopagoclient.model.Card
import com.chainreaction.mercadopagoclient.model.Token
import retrofit.Callback
import retrofit.http.Body
import retrofit.http.POST
import retrofit.http.Query

interface GatewayService {
    @POST("/card_tokens")
    fun getToken(
        @Query("public_key") public_key: String?,
        @Body card: Card?,
        callback: Callback<Token?>?
    )

    @POST("/card_tokens")
    fun getToken(
        @Query("public_key") public_key: String?,
        @Body card: Card?
    ): Token?
}